utils/python/CIME/XML/entry_id.py

```diff
@@ -86,6 +86,40 @@ def get_type_info(self, vid):
             return None
         else:
             return self._get_type_info(node)
+    
+    def _get_default(self, node):
+        default = self.get_optional_node("default_value", root=node)
+        if default is not None:
+            return default.text
+        else:
+            return None
+        
+    
+    # Get type description , expect child with tag "type" for node         
+    def _get_type (self, node):
+        type_node = self.get_optional_node("type", root=node)
+        if type_node is not None:
+            return type_node.text
+        else:
+            # Default to string
+            return "char"
+            
+    # Get description , expect child with tag "description" for parent node
+    def _get_description (self, node):
+        type_node = self.get_optional_node("desc", root=node)
+        if type_node is not None:
+            return type_node.text
+        else:
+            return None
+
+    # Get group , expect child with tag "group" for parent node        
+    def _get_group (self, node):
```

Comment:
group is no longer a field within each variable, instead it looks like:
'''
  <group id="external_tools">
    <entry id="DATA_ASSIMILATION" value="FALSE">
      <type>logical</type>
      <valid_values>TRUE,FALSE</valid_values>
      <desc> Run the external tool pointed to by DATA_ASSIMILATION_SCRIPT after the model run completes </desc>
    </entry>
    <entry id="DATA_ASSIMILATION_CYCLES" value="1">
      <type>integer</type>
      <valid_values/>
      <desc> Number of model run - data assimilation steps to complete </desc>
    </entry>
    <entry id="DATA_ASSIMILATION_SCRIPT" value="">
      <type>char</type>
      <valid_values/>
      <desc>External script to be run after model completion</desc>
    </entry>
  </group>
'''